### ISSUE TITLE
TorrentDownloadStation : error 160 Unable to retrieve queue and history items

### ISSUE THREAD

**Issue by sephrat** [Radarr/Radarr#2371]
**Description:**
I have Radarr installed on my NAS, and I use Download Station (namely Synology-DS in Radarr). 
The torrents are downloaded into some subfolder in /common/Vidéos/. 
Once a torrent is finished downloading (and only then), everything breaks down:
![error 160](https://user-images.githubusercontent.com/34862846/34390018-50ccf7a6-eb3d-11e7-9a42-3fd48cd38ba5.png)
This shouldn't be a permission issue, as I manage to manually add movies from that same folder. 
![Manual add](https://user-images.githubusercontent.com/34862846/34390284-27b85a34-eb3f-11e7-92c0-39b54226a123.jpg)
The activity feature does work fine - I'm able to see the download progress.

Please note that internally, the full path is /volume1/common/Vidéos (Synology infrastructure).
![diskstation - synology diskstation - google chrome](https://user-images.githubusercontent.com/34862846/34390108-f0d1bbec-eb3d-11e7-8053-ee8e28f06430.jpg)
![Internal path](https://user-images.githubusercontent.com/34862846/34390250-d5b5ec06-eb3e-11e7-95dc-18e172a03ba4.jpg)

**Radarr Version:** 0.2.0.910

**Logs:**
```
17-12-27 19:37:27.7|Debug|FileStationProxy|Trying to get info of /common
17-12-27 19:37:27.8|Error|FileStationProxy|Failed to get info of /common. Reason: 160 - Unknown error
17-12-27 19:37:27.9|Warn|SharedFolderResolver|Failed to get shared folder /common from Disk Station 127.0.0.1:5000

[v0.2.0.910] NzbDrone.Core.Download.Clients.DownloadClientException: Failed to get info of /common. Reason: 160 - Unknown error
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.DiskStationProxyBase.ProcessRequest[T] (NzbDrone.Common.Http.HttpRequestBuilder requestBuilder, System.String operation, NzbDrone.Core.Download.Clients.DownloadStation.DiskStationApi api, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x000a9] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\DiskStationProxyBase.cs:107 
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.DiskStationProxyBase.ProcessRequest[T] (NzbDrone.Common.Http.HttpRequestBuilder requestBuilder, System.String operation, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\DiskStationProxyBase.cs:66 
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.FileStationProxy.GetInfoFileOrDirectory (System.String path, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x0003d] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\FileStationProxy.cs:41 
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.FileStationProxy.GetSharedFolderMapping (System.String sharedFolder, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\FileStationProxy.cs:28 
  at NzbDrone.Core.Download.Clients.DownloadStation.SharedFolderResolver.GetPhysicalPath (NzbDrone.Common.Disk.OsPath sharedFolder, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\SharedFolderResolver.cs:33 

17-12-27 19:37:28.2|Warn|DownloadMonitoringService|Unable to retrieve queue and history items from Synology-DS

[v0.2.0.910] NzbDrone.Core.Download.Clients.DownloadClientException: Failed to get info of /common. Reason: 160 - Unknown error
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.DiskStationProxyBase.ProcessRequest[T] (NzbDrone.Common.Http.HttpRequestBuilder requestBuilder, System.String operation, NzbDrone.Core.Download.Clients.DownloadStation.DiskStationApi api, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x000a9] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\DiskStationProxyBase.cs:107 
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.DiskStationProxyBase.ProcessRequest[T] (NzbDrone.Common.Http.HttpRequestBuilder requestBuilder, System.String operation, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\DiskStationProxyBase.cs:66 
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.FileStationProxy.GetInfoFileOrDirectory (System.String path, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x0003d] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\FileStationProxy.cs:41 
  at NzbDrone.Core.Download.Clients.DownloadStation.Proxies.FileStationProxy.GetSharedFolderMapping (System.String sharedFolder, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\Proxies\FileStationProxy.cs:28 
  at NzbDrone.Core.Download.Clients.DownloadStation.SharedFolderResolver.GetPhysicalPath (NzbDrone.Common.Disk.OsPath sharedFolder, NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\DownloadStation\SharedFolderResolver.cs:33 

```

**Comment by sephrat** [comment]
For the record, it seems to occur in Sonarr as well (not tested myself): https://github.com/Sonarr/Sonarr/issues/1758

**Comment by JKetelaar** [comment]
I'm facing the same issue, has anyone found a solution to this yet?

**Comment by galli-leo** [comment]
Not really, since this isn't a documented error code, not much we can do.

**Comment by JKetelaar** [comment]
I managed to debug the error from my side and solve it (at least for me).
Here are the two solutions that made it work:

1. Make sure that the account accessing your Synology from Radarr/Sonarr doesn't just have download-station access. It also needs file-station access, as the API calls the file-station after the download is finished within the download-station.

2. Within the download client section I have added a Remote Path Mapping:
```
Host: host set as in the download client
Remote Path: `/volume1/your_path/on_the/synology` (`/volume1/video/Series/Downloads/` for me)
Local path: `/your_path/without/volume1` (`/video/Series/Downloads/` for me)
```

Hope this helps anyone!
